desci-models/src/trees/treeTools.ts

```diff
@@ -0,0 +1,291 @@
+    const parentPath = pathSplit.join("/");
+    const parent = pathToCompMap[parentPath];
+    if (parent && parent.type !== ResearchObjectComponentType.UNKNOWN) {
+      return parent;
+    }
+  }
+  return null;
+}
+
+//Convert IPFS tree to DriveObject tree V2
+export function convertIpfsTreeToDriveObjectTree(
+  tree: DriveObject[],
+  pathToCompMap: Record<DrivePath, ResearchObjectV1Component>,
+  pathToSizeMap: Record<DrivePath, number>
+) {
+  // tree = tree.filter((branch) => !FILTER_LIST.includes(branch.name)); // LEAVE THIS TO THE FRONTEND
+  tree.forEach((branch) => {
+    const fileDirBranch = branch as FileDir;
+    const neutralPath = neutralizePath(branch.path!);
+    branch.path = neutralPath;
+    const component = pathToCompMap[branch.path!];
+    const ancestorComponent: ResearchObjectV1Component | null =
+      getAncestorComponent(branch, pathToCompMap);
+    branch.componentType =
+      component?.type ||
+      ancestorComponent?.type ||
+      ResearchObjectComponentType.UNKNOWN;
+
+    if (component) {
+      const subtype =
+        "subtype" in component
+          ? (component["subtype"] as ResearchObjectComponentSubtypes)
+          : undefined;
+      if (subtype) branch.componentSubtype = subtype;
+    }
+    // useful for annotation insert on file tree under a code component for example (refer to component id later)
+    branch.componentId = component?.id || ancestorComponent?.id;
+    branch.accessStatus = fileDirBranch.published
+      ? AccessStatus.PUBLIC
+      : AccessStatus.PRIVATE;
+
+    //Determine partials
+    if (!fileDirBranch.published && branch.contains && branch.contains.length) {
+      const isPartial = hasPublic(branch);
+      if (isPartial) branch.accessStatus = AccessStatus.PARTIAL;
+    }
+
+    if (branch.external) branch.accessStatus = AccessStatus.EXTERNAL;
+
+    branch.metadata = inheritMetadata(branch.path, pathToCompMap);
+    branch.starred = component?.starred || false;
+    // branch.lastModified = formatDbDate(branch.lastModified) || tempDate; // LEAVE THIS TO FRONTEND
+    if (
+      branch.contains &&
+      branch.contains.length &&
+      branch.type === FileType.DIR
+    ) {
+      branch.size = pathToSizeMap[branch.path!] || 0;
+      branch.contains = convertIpfsTreeToDriveObjectTree(
+        branch.contains,
+        pathToCompMap,
+        pathToSizeMap
+      );
+    }
+  });
+  return tree;
+}
+
+export function hasPublic(tree: DriveObject): boolean {
+  return tree.contains!.some((fd) => {
+    const fdTyped = fd as FileDir;
+    if (fdTyped.published) return true;
+    if (fd.contains && fd.contains.length) return hasPublic(fd);
+    return false;
+  });
+}
+
+export function inheritMetadata(
+  path: DrivePath,
+  pathToCompMap: Record<DrivePath, ResearchObjectV1Component>
+) {
+  const comp = pathToCompMap[path];
+  if (comp) {
+    const specificMetadata = extractComponentMetadata(comp);
+    if (Object.keys(specificMetadata).length) return specificMetadata;
```

Comment:
Super nitpicky, but I'm personally not a fan of relying on truthy/falsy coercion for code flow :roll_eyes: 
```suggestion
    if (Object.keys(specificMetadata).length > 0) return specificMetadata;
```